### PR TITLE
 Fill in motivation section

### PR DESCRIPTION
**Pull request by danrubel**
This fills in the motivation section with content pulled from Paul's proposal doc

### CODE REVIEW

File: resources/type-system/flow-analysis.md [hunk 1]
```diff
@@ -15,7 +15,46 @@ reachability analysis.
 
 ## Motivation
 
-TODO
+The Dart language spec requires type promotion analysis to determine when
+a local variable is known to have a more specific type than its declared type.
+This is currently performed by the front end and the analyzer which perform additional
```
Comment (stereotype441) [inline resources/type-system/flow-analysis.md]: This is not really accurate--the "type propagation" phase that used to exist in the analyzer was removed with Dart 2.0.  At the moment the analyzer and the front end both implement the same type promotion rules, which match the spec.

File: resources/type-system/flow-analysis.md [hunk 2]
```diff
@@ -15,7 +15,46 @@ reachability analysis.
...
+proposal (see [tracking issue](https://github.com/dart-lang/language/issues/81)). We believe
+additional enhancement is necessary beyond this proposal to support NNBD (“non-nullability by default”).
+
+The language spec also requires a small amount of reachability analysis,
+to ensure control flow does not reach the end of a switch case.
+To support NNBD a more complete form of reachability analysis is necessary to detect
+when control flow “falls through” to the end of a function body and an implicit `return null;`,
+which would be an error in a function with a non-nullable return type.
+
+TODO: Links to proposals or issues requesting additional reachability analysis?
```
Comment (stereotype441) [inline resources/type-system/flow-analysis.md]: This is the issue I'm aware of: https://github.com/dart-lang/language/issues/139

File: resources/type-system/flow-analysis.md [hunk 3]
```diff
@@ -15,7 +15,46 @@ reachability analysis.
...
+
+I believe that all three kinds of analysis can be specified (and implemented) using a common framework,
+and that by doing so, we can make all three of them more sophisticated without a great deal of extra effort.
+This document outlines the formulation I have in mind, and illustrates the benefits through some examples.
+The design is inspired by the Wikipedia article on definite assignment analysis, and shares many of its concepts
+with the prototype implementation of definite assignment analysis in the analyzer, as well as the implementation
+of type promotion in the front end.  It also takes many ideas from prior work by Johnni Winther in Java,
+and it is similar to analysis that is currently done by the dart2js back-end.
+
+TODO: Is the above paragraph an "I" or "We" ?
```
Comment (stereotype441) [inline resources/type-system/flow-analysis.md]: Let's change it to "we".

File: resources/type-system/flow-analysis.md [hunk 4]
```diff
@@ -15,7 +15,46 @@ reachability analysis.
...
+
+This proposal comes in two possible variants: a single pass variant, which can be computed in a single pass
```
Comment (stereotype441) [inline resources/type-system/flow-analysis.md]: I presented two variants in my original doc, but based on feedback from Leaf I want this proposal to be a single coherent snapshot of the idea, without any variants the reader has to worry about (see the first bullet in the "Alternatives considered" section)

File: resources/type-system/flow-analysis.md (hunk 1 above)
Comment (danrubel) [reply resources/type-system/flow-analysis.md]: Ah! Missed that. Reworded.

File: resources/type-system/flow-analysis.md (hunk 2 above)
Comment (danrubel) [reply resources/type-system/flow-analysis.md]: Added.

File: resources/type-system/flow-analysis.md (hunk 3 above)
Comment (danrubel) [reply resources/type-system/flow-analysis.md]: Done.

File: resources/type-system/flow-analysis.md (hunk 4 above)
Comment (danrubel) [reply resources/type-system/flow-analysis.md]: In that case there is no need for this paragraph. Removed.

**Comment by danrubel** [comment]
Reworked. PTAL.

**Review by stereotype441 — APPROVED** [review]
lgtm, thank you!

**Pull request merged**